src/menu/ide_menu.rs

```diff
@@ -0,0 +1,1459 @@
+                + border_width)
+                .min(self.default_details.max_completion_width)
+                .max(self.default_details.min_completion_width)
+                .min(terminal_width.saturating_sub(min_description_width))
+                .max(3 + border_width); // Big enough to show "..."
+
+            let available_description_width = terminal_width
+                .saturating_sub(completion_width)
+                .min(self.default_details.max_description_width)
+                .max(self.default_details.min_description_width)
+                .min(terminal_width.saturating_sub(completion_width));
+
+            min_description_width = min_description_width.min(available_description_width);
+
+            let description_width = if let Some(description) = description {
+                self.description_dims(
+                    description,
+                    available_description_width,
+                    u16::MAX,
+                    min_description_width,
+                )
+                .0
+            } else {
+                0
+            };
+
+            let max_offset = terminal_width.saturating_sub(completion_width + description_width);
+
+            let description_offset = self.default_details.description_offset.min(max_offset);
+
+            self.working_details.completion_width = completion_width;
+            self.working_details.description_width = description_width;
+            self.working_details.description_offset = description_offset;
+            self.working_details.menu_width =
+                completion_width + description_offset + description_width;
+
+            let cursor_offset = self.default_details.cursor_offset;
+
+            self.working_details.description_is_right = match self.default_details.description_mode
+            {
+                DescriptionMode::Left => false,
+                DescriptionMode::Right => true,
+                DescriptionMode::PreferRight => {
+                    // if there is enough space to the right of the cursor, the description is shown on the right
+                    // otherwise it is shown on the left
+                    let potential_right_distance = (terminal_width as i16)
+                        .saturating_sub(
+                            cursor_pos as i16
+                                + cursor_offset
+                                + description_offset as i16
+                                + completion_width as i16,
+                        )
+                        .max(0) as u16;
+
+                    potential_right_distance >= description_width
+                }
+            };
+
+            let space_left = (if self.working_details.description_is_right {
+                cursor_pos as i16 + cursor_offset
+            } else {
+                (cursor_pos as i16 + cursor_offset)
+                    .saturating_sub(description_width as i16 + description_offset as i16)
```

Comment:
There needs to be a check here to see if there's any description at all to avoid adding `description_offset` in that case. Should fix the jumping issue (fingers crossed)